subsys/bluetooth/host/hci_core.c

```diff
@@ -1032,10 +1032,20 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	struct bt_conn *conn;
 	int err;
 
-	BT_DBG("status 0x%02x handle %u role %u %s", evt->status, handle,
-	       evt->role, bt_addr_le_str(&evt->peer_addr));
+	BT_DBG("status 0x%02x handle %u role %u adr:%s rpa:%s local_rpa:%s",
+			evt->status, handle, evt->role, bt_addr_le_str(&evt->peer_addr),
+			bt_addr_str(&evt->peer_rpa), bt_addr_str(&evt->local_rpa));
 
 #if defined(CONFIG_BT_SMP)
+	if (!bt_addr_cmp(&evt->peer_addr.a, BT_ADDR_NONE)) {
+		if (bt_addr_cmp(&evt->peer_rpa, BT_ADDR_NONE)) {
```

Comment:
Is the ST firmware actually sometimes giving your BT_ADDR_NONE as RPA? If not, I'd just leave out this secondary check.

Reply:
Totally agree with your point. I'm not familiar with the raw HCI packet format, thus it was easier for me to write code for the "parsed" HCI packet. Now I'm investigating which bytes to copy in the driver

Reply:
@pavlohamov Yeah, it is a bit more tricky to add it there.

Your buffer will consist of these:
struct bt_hci_evt_hdr
check if evt == BT_HCI_EVT_LE_META_EVENT
struct bt_hci_evt_le_meta_event
check if subevent == BT_HCI_EVT_LE_ENH_CONN_COMPLETE
struct bt_hci_evt_le_enh_conn_complete

Reply:
@joerchan Thank you for your directions. I've dumped the message content and understand that it could be simply cast'ed to bt_hci_evt_le_enh_conn_complete. Please take a look